deploy/docker/Dockerfile-multi-arch-distroless

```
@@ -0,0 +1,36 @@
+# source https://github.com/istio/istio/blob/master/docker/Dockerfile.distroless
+FROM gcr.io/distroless/static-debian11:latest-${TARGETARCH:-amd64} AS distroless_source
```

Remove the default value. We expect and require the TARGETARCH arg to be passed in - if not, we expect and require errors to occur. We do not want to blindly use "amd64" as the default here because if this is building other arches, this will be wrong. This currently hid the bug that I reported in my previous comment - the ARG is defined further down below, so when this line 1 is evaluated, the TARGETARCH isn't the one passed and is always defaulting to amd64, which is wrong for the other arch builds.

So we want this:
```suggestion
FROM gcr.io/distroless/static-debian11:latest-${TARGETARCH} AS distroless_source
```